pkg/postage/postagecontract/contract.go

```diff
@@ -305,7 +310,69 @@ func (c *postageContract) TopUpBatch(ctx context.Context, batchID []byte, topUpA
 	return ErrBatchTopUp
 }
 
+func (c *postageContract) ExpireBatches(ctx context.Context) error {
+	count, err := c.countExpiredBatches(ctx)
+	if err != nil {
+		return fmt.Errorf("count expired batches: %w", err)
+	}
+
+	for i := 0; i < int(count.Int64()); i += 100 {
```

Comment:
Casting `int64` to `int` can cause a problem on 32-bit architectures if the value represented is above the `int32` range. But the main problem here is the call to `count.Int64()`. There are methods such as `IsInt64()` that can determine if a number stored in `big.Int` can be represented as `int64`, which should be respected. To avoid assumptions, and thus possible truncation of ranges, I would suggest writing the for loop in the following way:
```Golang
for i := big.NewInt(0); i.Cmp(count) <= 0; i.Add(i, big.NewInt(100)) {
// ...
}
```

Reply:
this is no longer necessary as instead of count we are taking a bool from contract